components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/geo/service/GeoLocationProviderServiceImpl.java

```diff
@@ -260,24 +261,30 @@ public boolean saveGeoAlert(Alert alert, DeviceIdentifier identifier, String ale
                     "Unrecognized execution plan type: " + alertType + " while creating geo alert");
         }
 
-        //persist alert in registry
-        updateRegistry(getRegistryPath(alertType, identifier, alert.getQueryName()), identifier, content,
-                       options);
-
         //deploy alert into event processor
         EventProcessorAdminServiceStub eventprocessorStub = null;
         String action = (isUpdate ? "updating" : "creating");
         try {
+            String existingPlanName = null;
+            String executionPlanName = getExecutionPlanName(alertType, alert.getQueryName(),
+                    identifier.getId());
             eventprocessorStub = getEventProcessorAdminServiceStub();
             String parsedTemplate = parseTemplate(alertType, parseMap);
             String validationResponse = eventprocessorStub.validateExecutionPlan(parsedTemplate);
             if (validationResponse.equals("success")) {
                 if (isUpdate) {
-                    String executionPlanName = getExecutionPlanName(alertType, alert.getQueryName(),
-                                                                    identifier.getId());
                     eventprocessorStub.editActiveExecutionPlan(parsedTemplate, executionPlanName);
                 } else {
-                    eventprocessorStub.deployExecutionPlan(parsedTemplate);
+                    try {
+                        existingPlanName = eventprocessorStub.getActiveExecutionPlan(executionPlanName);
+                        if (existingPlanName.contains(executionPlanName)) {
+                            throw new AlertAlreadyExist("Execution plan with this name already exists");
```

Comment:
This message should contain the executionPlanName. So message should be as follows.
"Execution plan already exists with name '" + executionPlanName + "'"